anta/tests/path_selection.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""Test functions related to various router path-selection settings."""
+
+# Mypy does not understand AntaTest.Input typing
+# mypy: disable-error-code=attr-defined
+from __future__ import annotations
+
+from ipaddress import IPv4Address
+from typing import ClassVar
+
+from pydantic import BaseModel
+
+from anta.models import AntaCommand, AntaTemplate, AntaTest
+from anta.tools import get_value
+
+
+class VerifyRouterPathsHealth(AntaTest):
+    """
+    Verifies the route and telemetry state of all paths under router path-selection.
+
+    The expected states are 'IPsec established', 'Resolved' for route and 'active' for telemetry.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all paths under router path-selection have their route state as either 'IPsec established' or 'Resolved'
+               and their telemetry state as 'active'.
+    * Failure: The test will fail if router path-selection is not configured, any path's route state is not 'IPsec established' or 'Resolved',
```

Comment:
Missing telemetry active check in the docstring  (to be added) and reword like this:

```suggestion
    * Failure: The test will fail if router path-selection is not configured or if any path state is not 'IPsec established' or 'Resolved',
```